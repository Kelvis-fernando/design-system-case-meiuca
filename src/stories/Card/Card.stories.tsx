import type { Meta, StoryObj } from "@storybook/react";
import { tokens } from "../../../tokens/index";
import { Card } from "./Card";
import { Heading } from "../Heading/Heading";
import { Subtitle } from "../Subtitle/Subtitle";
import { Button } from "../Button/Button";
import { Paragraph } from "../Paragraph/Paragraph";
import { CardContent } from "./Card.style";

const meta = {
  title: "Components/Card",
  component: Card,
  tags: ["autodocs"],
} satisfies Meta<typeof Card>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: (
      <>
        <img
          src="https://secure-static.vans.com.br/medias/sys_master/vans/vans/h66/h49/h00/h00/11319247536158/1002001070011U-01-BASEIMAGE-Hires.jpg"
          alt="tenis"
          style={{ width: "100%" }}
        />
        <div style={CardContent}>
          <Heading level={1} text={"Tenis"} />
          <Subtitle text={"Vans Old skool"} />
          <Paragraph text="Feito e entregue no Brasil" />
          <Button label={"Comprar"} />
        </div>
      </>
    ),
    width: 150,
  },
};
