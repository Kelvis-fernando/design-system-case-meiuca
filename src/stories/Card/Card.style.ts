import React from "react";
import { tokens } from "../../../tokens";

export const CardContent: React.CSSProperties = {
  padding: tokens.spacings.spacing.size.xxxs.value,
  display: "flex",
  flexDirection: "column",
  textAlign: "center",
  justifyContent: "center",
};
