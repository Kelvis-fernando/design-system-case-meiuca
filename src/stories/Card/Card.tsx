import { CardProps } from "../../types/CardProps";
import { Shape } from "../Shape/Shape";

export const Card: React.FC<CardProps> = ({
  width,
  height,
  children,
  ...props
}) => {
  return (
    <Shape width={width} height={height} {...props}>
      {children}
    </Shape>
  );
};
