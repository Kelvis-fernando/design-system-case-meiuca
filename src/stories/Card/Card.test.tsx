import React from "react";
import { render, screen } from "@testing-library/react";
import { Card } from "./Card";

describe("Card", () => {
  test("renders with correct width and height", () => {
    const width = 200;
    const height = 300;

    const { container } = render(
      <Card
        width={width}
        height={height}
        data-testid="card"
        children={undefined}
      />
    );

    const cardElement = screen.getByTestId("card");

    expect(cardElement).toHaveStyle(`width: ${width}px`);
    expect(cardElement).toHaveStyle(`height: ${height}px`);
    expect(container).toMatchSnapshot();
  });

  test("renders children content", () => {
    const children = "Card content";

    render(<Card>{children}</Card>);

    const contentElement = screen.getByText(children);

    expect(contentElement).toBeInTheDocument();
  });
});
