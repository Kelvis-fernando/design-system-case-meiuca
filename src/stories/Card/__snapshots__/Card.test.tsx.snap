// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders with correct width and height 1`] = `
<div>
  <div
    data-testid="card"
    style="background-color: rgb(255, 255, 255); box-shadow: 0px 2px 4px #A6AAAF; width: 200px; height: 300px; border-radius: 8px;"
  />
</div>
`;
